lib/internal/bootstrap/pre_execution.js

```diff
@@ -332,6 +335,61 @@ function initializeDeprecations() {
     enumerable: false,
     configurable: true
   });
+
+  let warnedForProto = false;
+  const ObjectPrototype = ObjectGetPrototypeOf({});
+  function warnOnProto() {
+    if (!warnedForProto) {
+      warnedForProto = true;
+      process.emitWarning('usage of Object.prototype.__* properties are a' +
+        ' common security concern, use static methods on Object instead');
+    }
+  }
+  const legacy = [
+    '__proto__',
```

Comment:
Is it going to warn for `const x = { __proto__: null, prop: 42 }` ?

Reply:
No, literals are not firing accessors, they use define semantics; __proto__ is a special syntax not a reflective API call. https://tc39.es/ecma262/#sec-runtime-semantics-propertydefinitionevaluation